go/ocr2/decryptionplugin/decryption.go

```diff
@@ -269,11 +271,20 @@ func (dp *decryptionPlugin) Report(ctx context.Context, ts types.ReportTimestamp
 			continue
 		}
 
-		// OCR2.0 guaranties 2f+1 observations are from distinct oracles
+		// This is a sanity check.
+		// OCR2.0 guaranties 2f+1 observations are from distinct oracles.
 		// which guaranties f+1 valid observations and, hence, f+1 valid decryption shares.
-		// Therefore, here it is guaranteed that len(decrShares) > f.
+		// Therefore, here it should be guaranteed that len(decrShares) > f.
+		if len(decrShares) < dp.genericConfig.F+1 {
```

Comment:
if possible, add a unit test case to cover this branch

Reply:
I believe this already covers it, it is actually the reason I added the sanity check:
https://github.com/smartcontractkit/tdh2/blob/2cbe7525a98f96b8e47f41a3053203f72de7ec29/go/ocr2/decryptionplugin/decryption_test.go#L642